composeApp/src/commonMain/kotlin/com/swmansion/routinetracker/viewmodel/SettingsViewModel.kt

```diff
@@ -31,27 +55,204 @@ class SettingsViewModel(
                     unspecifiedReminderMinute = p.unspecifiedReminderMinute,
                 )
             }
-            .stateIn(
-                scope = viewModelScope,
-                started = SharingStarted.WhileSubscribed(5_000),
-                initialValue = SettingsUiState(),
+            .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5_000), SettingsUiState())
+
+    private fun cancel(uuid: String) = alarmeeService.local.cancel(uuid)
+
+    private fun specifiedUuid(routineId: Long, dayOfWeek: Int) =
+        "$UUID_SPECIFIED-$routineId-$dayOfWeek"
+
+    private suspend fun cancelAllSpecifiedReminders() {
+        val routines = dataRepository.getAllRoutinesWithTasks().first()
+        routines
+            .map { it.routine }
+            .filter { it.time != null }
+            .forEach { routine ->
+                dataRepository.getRecurrencesForRoutine(routine.id).forEach { rec ->
+                    cancel(specifiedUuid(routine.id, rec.dayOfWeek))
+                }
+            }
+    }
+
+    @OptIn(ExperimentalTime::class)
+    fun scheduleSpecifiedReminderForRoutine(
+        routine: Routine,
+        recurrences: List<RoutineRecurrence>,
+        hour: Int,
+        minute: Int,
+    ) {
+        val pref = uiState.value
+        val offset = getReminderOffset(pref.specifiedSelected)
+        val nowInstant = getCurrentInstant()
+        val tz = getCurrentTimeZone()
+
+        recurrences.forEach { rec ->
+            scheduleRoutineReminder(routine, rec, hour, minute, offset, nowInstant, tz)
+        }
+    }
+
+    @OptIn(ExperimentalTime::class)
+    fun scheduleSpecifiedReminder() {
+        val pref = uiState.value
+        val offset = getReminderOffset(pref.specifiedSelected)
+
+        viewModelScope.launch {
+            val routines = dataRepository.getAllRoutinesWithTasks().first()
+            val nowInstant = getCurrentInstant()
+            val tz = getCurrentTimeZone()
+
+            routines
+                .map { it.routine }
+                .filter { it.time != null }
+                .forEach { routine ->
+                    val (hour, minute) = parseHourMinute(routine.time!!)
```

Comment:
We can avoid the not-null assertion:

```suggestion
                .forEach { routine ->
                    if (routine.time == null) return@forEach
                    val (hour, minute) = parseHourMinute(routine.time)
```